plutus-ledger-constraints/src/Ledger/Constraints/OnChain/V2.hs

```diff
@@ -106,7 +106,7 @@ checkTxConstraint ctx@ScriptContext{scriptContextTxInfo} = \case
         in
         traceIfFalse "L7" -- "Public key output not spent"
         $ maybe False (isNoOutputDatum . txOutDatum . txInInfoResolved) (PV2.findTxInByTxOutRef txOutRef scriptContextTxInfo)
-    MustSpendScriptOutput txOutRef _ ->
+    MustSpendScriptOutput txOutRef _ _ ->
```

Comment:
Don't we need a check that the provided `TxOutRef` is part of the reference inputs?

Reply:
I don't think we need to perform this check because the ledger enforces datum witness. If important, developer can use `MustReferenceOutput`.

Reply:
I like that idea @james-iohk, but sadly that would create two reference inputs. So I added the check.

Reply:
I meant they can use the onchain constraint only. But I think what you've done keeps a simpler user experience. 
I didn't think an input could be referenced twice :thinking: